### PR TITLE
feat: estimate message fee schema changes

### PR DESCRIPTION
**Pull request by Mirko-von-Leipzig**
This PR bootstraps initial v0.4 RPC support and updates `starknet_estimateMessageFee` to match the latest schema.

I've bootstrapepd v0.4 support by just copying across the v0.3 methods. 

`starknet_estimateMessageFee` is split into v0.3 and v0.4 by mapping the latters input changes to the formers.

### CODE REVIEW

File: crates/rpc/src/v04/method/estimate_message_fee.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    let entry_point_selector = EntryPoint(
+        Felt::from_be_slice(input.message.from_address.0.as_bytes())
+            .expect("H160 bytes will not overflow felt"),
+    );
+    let input = crate::v03::method::estimate_message_fee::EstimateMessageFeeInput {
+        message: FunctionCall {
+            contract_address: input.message.to_address,
+            entry_point_selector,
+            calldata: input.message.payload,
+        },
```
Comment (Mirko-von-Leipzig) [inline crates/rpc/src/v04/method/estimate_message_fee.rs:36]: I'm not convinced this is correct?

Comment (kkovaacs) [reply crates/rpc/src/v04/method/estimate_message_fee.rs:36]: It looks OK on the first blink. Would be great to add a test though that makes sure we got this mapping right.

File: crates/rpc/src/v04/method/estimate_message_fee.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+use crate::context::RpcContext;
+use crate::v02::method::call::FunctionCall;
+use crate::v02::types::reply::FeeEstimate;
+use crate::v03::method::estimate_message_fee::EstimateMessageFeeError;
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct EstimateMessageFeeInput {
+    message: MsgFromL1,
+    sender_address: EthereumAddress,
```
Comment (kkovaacs) [inline crates/rpc/src/v04/method/estimate_message_fee.rs:12]: This is redundant (`sender_address` is now in `MsgFromL1` as `from_address`).

File: crates/rpc/src/v04/method/estimate_message_fee.rs (hunk 1 above)
Comment (kkovaacs) [reply crates/rpc/src/v04/method/estimate_message_fee.rs:36]: `entry_point_selector` has been added, so maybe it's time to re-visit this PR.

Comment (Mirko-von-Leipzig) [reply crates/rpc/src/v04/method/estimate_message_fee.rs]: Fixed

File: crates/rpc/src/v04/method/estimate_message_fee.rs (hunk 2 above)
Comment (Mirko-von-Leipzig) [reply crates/rpc/src/v04/method/estimate_message_fee.rs]: Fixed

**Comment by Mirko-von-Leipzig** [comment]
Updated with the new rpc updates; makes much more sense now.

**Review by kkovaacs — APPROVED** [review]

**Pull request merged**